module/data/item/spell.mjs

```diff
@@ -79,9 +78,9 @@ export default class SpellData extends SystemDataModel.mixin(
    * Migrate the component object to be 'properties' instead.
    * @param {object} source  The candidate source data from which the model will be constructed.
    */
-  static #migrateComponentData(source) {
-    if ( !source.components || ("properties" in source)) return;
-    source.properties = filteredKeys(source.components);
+  static _migrateComponentData(source) {
+    const components = filteredKeys(source.system.components ?? {});
```

Comment:
It's possible `system` might not be in `source`. If only the Item's `name` is updated, for example.

```suggestion
    const components = filteredKeys(source.system?.components ?? {});
```